### ISSUE TITLE
Arc Palette Suggestions

### ISSUE THREAD

**Issue by Dhyfer1** [neurokitti/Arc_Palette#7]
When I run the .bat file I get this error

![Screenshot 2024-08-12 224227.png](https://github.com/user-attachments/assets/06169fe4-d980-4531-b818-23a22ffe4016)

and Arc_Palette does not appear, what is the solution?

**Comment by StudioMovieGirl** [comment]
Use the arc_pallet.bat file I the v1.2 release

**Comment by StudioMovieGirl** [comment]
You don't have git installed so we changed how the script works.

**Comment by Andrew-J-Larson** [comment]
@Dhyfer1 try the latest version at https://github.com/neurokitti/Arc_Palette/releases/tag/v1.5:
- Download the `WINDOWS_arc_palette_full_setup.bat` file

**Comment by Dhyfer1** [comment]
@Andrew-J-Larson I have downloaded the executable for the latest version, but it doesn't work

![Screenshot 2024-08-13 221843.png](https://github.com/user-attachments/assets/7fdf5dec-2049-4fa4-b3a7-f4359c262001)

Additionally, Windows Defender indicates that the executable has a trojan, and [this is the report by VirusTotal](https://www.virustotal.com/gui/file/ad0bc93d064396ebd1031fc328131740e87589f0e2fa84180001c4505a92f7f2/detection) 

![Screenshot 2024-08-13 221227.png](https://github.com/user-attachments/assets/bfca7991-8d8b-47ee-bd72-d0d8573afde0)

Finally, since Arc_Palette isn't working, it creates a file called debug.log, which only has this line:

`[0813/221108.477: ERROR:crashpad_client_win_cc(810)] not connected`

Every time I try to apply a palette, a new line appears in the debug.log file, for example: 

>`[0813/221108.477: ERROR:crashpad_client_win_cc(810)] not connected`

>`[0813/222715.957: ERROR:crashpad_client_win_cc(810)] not connected`

**Comment by Andrew-J-Larson** [comment]
@Dhyfer1 yeah I was looking into it, after some research, I think it's only because of setting the `--icon` for the app when building with pyinstaller (weird yeah I know) that it's getting triggered in antiviruses for some reason. 

Might just need to continue further builds without the icon on Windows. (but this won't affect using the icon inside the actual application itself)

While Defender is still blocking the application, I don't imagine that it probably won't work right until it's unblocked, that's what the VirusTotal reports are there for to check against (and report false positives).

I probably won't be able to generate a new build of the exe until later (16 hours ish from now) when I'm home.

**Comment by Andrew-J-Larson** [comment]
Try to use the setup and then run script in with the `resources\Windows` files for now and see if those work.

**Comment by Dhyfer1** [comment]
@Andrew-J-Larson My problem was solved when I set a theme in the main space, then used Arc_Palette (version 1.9) and now it works. So it is necessary to theme the space before using Arc_Palette, that should be mentioned in the Readme.

![Screenshot 2024-08-14 080423.png](https://github.com/user-attachments/assets/3b41ea08-adfc-4332-af21-a9cf9d9636af)

 I just did what you mentioned before in another issue: https://github.com/neurokitti/Arc_Palette/issues/10#issuecomment-2285550643

Thanks for that, lastly as a feature requirement I would like Arc_Palette to be able to have a dark theme, as well as adding hex or RGB values.

Edit: It only works in the first space. I have created a second space, set a theme for that second space, run Arc_Palette but there are no visible changes.

**Comment by Dhyfer1** [comment]
@Andrew-J-Larson My problem was solved when I set a theme in Arc settings, then used Arc_Palette and it works in my main space.

![Screenshot 2024-08-14 080423.png](https://github.com/user-attachments/assets/3b41ea08-adfc-4332-af21-a9cf9d9636af)

 I just did what you mentioned before in another issue: #issuecomment-2285550643_

Thanks for that, lastly as a feature requirement I would like Arc _Palette to be able to have a dark theme,as well as adding hex or RGB values and be able to eliminate colors

**Comment by Andrew-J-Larson** [comment]
> Edit: It only works in the first space. I have created a second space, set a theme for that second space, run Arc_Palette but there are no visible changes.

Yeah, that's not yet fixed, @Dhyfer1 I'll be fixing that after the recode, plus I'm also at work right now, so I'm going to be occupied until I get home to fix it.

**Comment by Andrew-J-Larson** [comment]
Also, you need to close Arc after setting that theme, so that it updates the changes to the settings file that Arc Palette reads from.

**Comment by Dhyfer1** [comment]
> Also, you need to close Arc after setting that theme, so that it updates the changes to the settings file that Arc Palette reads from.

It doesn't work. I created a new space, gave it a name and a theme, then closed Arc. Then I open Arc_Palette, picked a couple of colors, checked the box to restart Arc, and then the ✓ to apply it. But after reopening Arc, there is no gradient in the second space.

I hope you can fix it

**Comment by Andrew-J-Larson** [comment]
> > Also, you need to close Arc after setting that theme, so that it updates the changes to the settings file that Arc Palette reads from.
> 
> It doesn't work. I created a new space, gave it a name and a theme, then closed Arc. Then I open Arc_Palette, picked a couple of colors, checked the box to restart Arc, and then the ✓ to apply it. But after reopening Arc, there is no gradient in the second space.
> 
> I hope you can fix it

Are you opening Arc Palette during the creation of the new space? What happens if you have Arc Palette completely closed, create that space in Arc, and then open Arc Palette to try and set a new theme?

Does that follow with the same results or not?

**Comment by Dhyfer1** [comment]
> Are you opening Arc Palette during the creation of the new space? 

No

> What happens if you have Arc Palette completely closed, create that space in Arc, and then open Arc Palette to try and set a new theme?
> 
> Does that follow with the same results or not?

I did what you said, but the second space does not apply the palette. I have also tried several combinations.

- Arc Palette is not open
- Open Arc and I create a new space with a name and a theme
- I open Arc Palette, select some colors and apply the palette 
- When Arc reopens, no changes are seen

Another combination 

- Arc Palette is not open
- Open Arc and I create a new space with a name and a theme
- I close Arc
- I open Arc Palette, select some colors and apply the palette 
- When Arc reopens, no changes are seen

Another combination: Same as in the first combination, but this time Arc Palette is open

Another combination: Same as in the second combination, but this time Arc Palette is open

**Comment by Andrew-J-Larson** [comment]
I have a sneaking suspicion I know what's happening, I'll take a look and see what I can do.

**Comment by Dhyfer1** [comment]
Ok, I look forward to an update about this.

**Comment by Andrew-J-Larson** [comment]
@Dhyfer1 try the latest update: https://github.com/neurokitti/Arc_Palette/releases/tag/v0.2.1-beta

Error should be fixed, and should also have less struggle with antiviruses

**Comment by Dhyfer1** [comment]
Thanks for the new version, looks great! ☺️

For practical purposes, I will refer to Arc Browser as AB and Arc Palette as AP 

By the way, when I told you this: https://github.com/neurokitti/Arc_Palette/issues/7#issuecomment-2289797018, I was using AB version 1.14.0, in that version AP (version 0.1.9) I could not apply a palette to a second space. 

Then, as if by magic, AP started to work on more than one space and also displayed tabs for each space after AB was updated to version 1.15 which was released yesterday.

You can see in this screenshot version 0.1.9 and version 0.2.1, both with several tabs. The last version did not jump any alert in Windows Defender. :D

![Untitled.png](https://github.com/user-attachments/assets/e5b53ebb-49ab-4cf6-a924-f1b2e9cebae1)

I have some suggestions for AP:

- In the screenshot you can see that the circles are clean in their inner and outer contour (well rounded) as shown in the window to edit the space theme in AB, in AP the circles look pixelated in their inner and outer contour, I would like them to be well rounded as in AB.

- When more than one tab appeared in AP (v0.1.9), and since I had several open spaces, I wanted to test if the palettes I put to each space could all be applied at the same time, but it didn't work, I could only apply one palette to one space at a time. It would be great if multiple palettes could be applied to multiple spaces at once.

- I think a button to reset AP would be useful. Example: If I have AP open with two tabs (space 1 and 2) and I also have AB open with two spaces, and then I create spaces 3, 4, 5 and 6, these 4 added spaces do not appear automatically in AP, for that I would have to reopen it (go to the location of the executable and reopen it). To save time, a button to restart AP would be useful, if there is such a button, AP reopens by itself showing the newly added slots.

- Spaces can have names: Study, Work, etc. It would be interesting if AP could display the names of the spaces instead of Space 1, Space 2, Space 3, etc.

- It would also be useful to add a color using values in hex or RGB format.

These are some ideas, I hope you take them into consideration.

**Comment by valk-ryx** [comment]
Thank you for the suggestions, I've renamed your issue and reopened it in order to make it easier to navigate to.

**Comment by Andrew-J-Larson** [comment]
Yes, I'm already working on a few of those things mentioned.

**Comment by Dhyfer1** [comment]
Great! I'm looking forward to seeing what's new in the next version, I'm hoping for something like what I said. 😁

**Comment by Andrew-J-Larson** [comment]
> * In the screenshot you can see that the circles are clean in their inner and outer contour (well rounded) as shown in the window to edit the space theme in AB, in AP the circles look pixelated in their inner and outer contour, I would like them to be well rounded as in AB.

The current graphics library we use doesn't have anti-aliasing, so we would have to switch to something entirely different to solve that problem, not sure if that's worth it at this time.

> * When more than one tab appeared in AP (v0.1.9), and since I had several open spaces, I wanted to test if the palettes I put to each space could all be applied at the same time, but it didn't work, I could only apply one palette to one space at a time. It would be great if multiple palettes could be applied to multiple spaces at once.

Right now there is a rewrite of the API accessing the config file, so that might be something we can take a look at afterwards.

> * I think a button to reset AP would be useful. Example: If I have AP open with two tabs (space 1 and 2) and I also have AB open with two spaces, and then I create spaces 3, 4, 5 and 6, these 4 added spaces do not appear automatically in AP, for that I would have to reopen it (go to the location of the executable and reopen it). To save time, a button to restart AP would be useful, if there is such a button, AP reopens by itself showing the newly added slots.

Yes, I've already brought up the idea of a "Reload" or "Reread" config button, for this very purpose.

> * Spaces can have names: Study, Work, etc. It would be interesting if AP could display the names of the spaces instead of Space 1, Space 2, Space 3, etc.

This is actually something I've already written a fix for, and pushing out sometime shortly.

> * It would also be useful to add a color using values in hex or RGB format.

This sounds like an easy one when the time allows @valk-ryx

**Comment by valk-ryx** [comment]
> Great! I'm looking forward to seeing what's new in the next version, I'm hoping for something like what I said. 😁

A few of your suggestions will most likely be added but we are mainly focusing on backend things at the moment so expect it in a few updates time.

**Comment by valk-ryx** [comment]
>> * This sounds like an easy one when the time allows

Yeah, originally it was going to be an issue since we didn't have a proper colorspace but it should work now, the main issue we might have is getting the circle to move to the correct place on the dot pad.

**Comment by Dhyfer1** [comment]
Thanks for checking out my suggestions, that's all I've come up with so far. 😅

**Comment by Dhyfer1** [comment]
> A few of your suggestions will most likely be added but we are mainly focusing on backend things at the moment so expect it in a few updates time.

Okay, the most important thing for me is that they are taken into account, so I will be watching for new changes in the next versions.

**Comment by Dhyfer1** [comment]
By the way, this is another suggestion that I forgot to add to the list, and that is to be able to change the gradient orientation: linear, angular, radial, conic etc as you can do in an image editor.

**Comment by Andrew-J-Larson** [comment]
> > > * This sounds like an easy one when the time allows
> 
> Yeah, originally it was going to be an issue since we didn't have a proper colorspace but it should work now, the main issue we might have is getting the circle to move to the correct place on the dot pad.

If we change it from instead of reading values directly from the canvas to instead logging the values directly to variable in the program, then we can have it approx. place the circle in the area where it should be on the canvas

**Comment by Andrew-J-Larson** [comment]
Here's a little update about the requests in this ticket:
- Original issue with `Git` was solved by not using it in scripts/creating executables for the program
- For the antialiasing, we could potentially 'mimic' what feels like anti-aliasing, but it wouldn't be actual anti-aliasing
- Still waiting on the backend API to be updated before we implement saving of all spaces at the same time (instead of one at a time)
- For the button idea of reloading Arc settings into Arc Palette, also waiting on the backend API to be updated before making any changes
- The update for Space names to show up instead of generic numbering is out now: https://github.com/neurokitti/Arc_Palette/releases/tag/v0.2.4-beta
- Unsure of what progress has been made on advanced color picking (from manually entered hex or RGB values)
- Unsure if changing gradient orientation/angle is possible at this time, but if it is, we'll fine out after the backend API is updated

**Comment by Dhyfer1** [comment]
Thanks for the update, I see some changes I mentioned before, such as the tabs now have the name of the space, 🥳 besides the circles are still pixelated, but this time a little less. I understand that this is due to 'mimic' anti-aliasing, I'm sure true anti-aliasing could solve that problem. 

I also see the improvement in the dark theme a little better, and the icon in the taskbar looks very nice.👌

Also I see that many of the missing changes I mentioned still require the API backend, I hope this API will be updated soon to add more changes for Arc Palette.

**Comment by Andrew-J-Larson** [comment]
> I understand that this is due to 'mimic' anti-aliasing, I'm sure true anti-aliasing could solve that problem.

No mimic-ing of anti-aliasing has been implemented yet, as we still have other things we need to work out in the code. Only thing I did was change the color to be darker for the outer ring (makes the pixelated look less noticeable yes).

**Comment by Dhyfer1** [comment]
Oh, I understand. I'm sure that with anti-aliasing it should look much better.
